addons/website_sale/models/mail_compose_message.py

```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import datetime
+from odoo import api, fields, models
+
+class MailComposeMessage(models.TransientModel):
+    _inherit = 'mail.compose.message'
+
+    @api.multi
+    def send_mail(self, auto_commit=False):
```

Comment:
Well what you are trying to do, if I understand well, is a bit tricky :) . I suppose you want to flag orders you just mailed that were abandoned carts. Several things here

1/ code optimization: we should avoid doing
```
records = self.search([])
for rec in records:
  rec.foo = bar
```
We should update records in batch and write on all of them at once
```
records = self.search([])
records.write({'foo': 'bar})
```
Note that this can be simplified into self.search([]).write({}), because you can pipe some recordsets methods.

2/ actually overriding the mail composer is not a good practice. You should look at mail.thread class and specifically at its message_post method. This involves inheritance, abstract classes, the chatter mecanism and a lot of funny stuff. I bet @rim-odoo will be glad to help you :p .

3/ mail_post_autofollow should be used only in some cases, but this is a bit complicated to explain here. First thing is probably to try to override message_post and maybe manually use message_subscribe (hahaaa, another key word).

A lot of key words in this comment. Be strong.